src/framework/preferences/js/Builder.js

```diff
@@ -196,12 +196,22 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
     };
 
     /*
-     * A one-stop-shop function to build and instantiate a prefsEditor from a schema.
+     * A one-stop-shop component to build and instantiate a prefsEditor from a schema.
      */
-    fluid.prefs.create = function (container, options) {
-        options = options || {};
-        var builder = fluid.prefs.builder(options.build);
-        return fluid.invokeGlobalFunction(builder.options.assembledPrefsEditorGrade, [container, options.prefsEditor]);
-    };
+    fluid.defaults("fluid.prefs.create", {
+        gradeNames: ["fluid.viewComponent"],
+        builder: {}, // options provided by users
+        prefsEditor: {}, // options provided by users
+        members: {
+            constructedGrades: "@expand:fluid.prefs.builder({that}.options.build)"
+        },
+        components: {
+            prefsEditor: {
+                type: "{fluid.prefs.create}.constructedGrades.options.assembledPrefsEditorGrade",
```

Comment:
This is a better approach than invoker + dynamic grade - as our docs suggest, "raw dynamic grades" are a last-ditch technique. 
http://docs.fluidproject.org/infusion/development/ComponentGrades.html#raw-dynamic-grades